buildSrc/src/main/java/org/opensearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -195,16 +201,22 @@ private static void setupDownloadServiceRepo(Project project) {
         if (project.getRepositories().findByName(DOWNLOAD_REPO_NAME) != null) {
             return;
         }
-        addIvyRepo(
-            project,
-            DOWNLOAD_REPO_NAME,
-            "https://artifacts.opensearch.org",
-            FAKE_IVY_GROUP,
-            "/releases" + RELEASE_PATTERN_LAYOUT,
-            "/release-candidates" + RELEASE_PATTERN_LAYOUT
-        );
-
-        addIvyRepo(project, SNAPSHOT_REPO_NAME, "https://artifacts.opensearch.org", FAKE_SNAPSHOT_IVY_GROUP, SNAPSHOT_PATTERN_LAYOUT);
+        // checks if custom Url has been passed by user from plugins
+        if (customUrl != null && !customUrl.isEmpty()) {
+            addIvyRepo(project, DOWNLOAD_REPO_NAME, customUrl, FAKE_IVY_GROUP, "");
+            addIvyRepo(project, SNAPSHOT_REPO_NAME, customUrl, FAKE_SNAPSHOT_IVY_GROUP, "");
+            customUrl = null;
```

Comment:
Why does this URL get reset?

Reply:
When I was testing in local, if I don't reset the customUrl to null it is getting cached by gradle and when someother plugin/method which don't want to use customUrl calls this method setupDownloadServiceRepo(), the customurl value is still being set to old value and condition in if block will be true and executed. So, to avoid this I had to reset the customUrl value. I observed when a plugin individually is tested against Opensearch it executes this method setupDownloadServiceRepo() only once.

Reply:
You have added a static property, so when you set it, it gets set globally, and forever. You need to figure out a way of how to set it only for the scope/instance that you are interested in instead.

Reply:
Yes I thought about it, but there is certain way the parameter is being passed to DistributionDownloadPlugin.java which makes to use static variable. 

--> First when plugin in ran against Opensearch it calls "TestClustersPlugin.java", here a call to DistributionDownoadPlugin.java is happening only once with project as parameter.

--> But Project values are still unset when a call to distributionDownloadPlugin.java is happening. So, I currently don't have custom URL passed by user from plugin. This file uses afterEvaluate() method which is listener interface and make some methods of file execute after project is evaluated.

--> Custom URL is passed to OpenSearchCluster.java using a setter method. This file is called by TestClustersPlugin.java after calling DistributionDownloadPlugin.java. So by the time I have custom URL from plugin, call to DistributionDownloadPlugin.java is already done.

--> If I try to create another instance of DistributionDownloadPlugin.java  after getting custom URL from plugin with same project as parameter, it throws an error that "this project is already present".

--> So, the only option I had to send custom URL to DistributionDownloadPlugin.java was to use a static setter and call it by it's class name.

--> I have tried different ways to avoid setting a static variable, but they were failing. I will try asking other developers for suggestions to avoid it.